.github/workflows/test.yml

```diff
@@ -40,14 +34,59 @@ jobs:
           user_service_endpoint: ${{ secrets.USER_SERVICE_ENDPOINT }}
           acl_provider_endpoint: ${{ secrets.ACL_PROVIDER_ENDPOINT }}
         run: lerna run test
+  
+  dep-check:
+    needs: install-deps
+    runs-on: [ubuntu-latest]
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/restore-cache
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/build-essential
       - name: Dependency Checks
         run: npm run depcheck
+  
+  prettier:
+    needs: install-deps
+    runs-on: [ubuntu-latest]
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/restore-cache
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/build-essential
       - name: Prettier
         run: npm run prettier:check
+  
+  linting:
+    needs: install-deps
+    runs-on: [ubuntu-latest]
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/restore-cache
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/build-essential
       - name: Linting
         run: lerna run lint
+
+  audit:
+    needs: install-deps
+    runs-on: [ubuntu-latest]
+    steps: 
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/restore-cache
+      - uses: ./.github/actions/setup-node
+      - uses: ./.github/actions/build-essential
       - name: Audit
         run: lerna run audit
+
+  e2e-tests:
+    needs: install-deps
```

Comment:
❓ Do parallelised steps run on the same machine ?
I'm wondering if the dependencies would be installed for this job or any other jobs.

Reply:
No they don't, each job is on a separate runner. 
So first the install step happens, the node modules are cached, then all the other jobs start at the same time. In each of them the cache is restored, and the tests happen.

Reply:
I see, so they don't actually have to install anything themselves, good 👍